tools/llm_bench/task/speech_to_text_generation.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2023-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+import os
+import time
+import datetime
+import numpy as np
+from pathlib import Path
+import hashlib
+import logging as log
+import llm_bench_utils
+import llm_bench_utils.model_utils as model_utils
+import llm_bench_utils.metrics_print as metrics_print
+import llm_bench_utils.gen_output_data as gen_output_data
+import llm_bench_utils.parse_json_data as parse_json_data
+
+FW_UTILS = {'pt': llm_bench_utils.pt_utils, 'ov': llm_bench_utils.ov_utils}
+
+
+def run_speech_2_txt_generation(raw_speech, pipe, args, num, md5_list, speech_id,
+                                iter_data_list, mem_consumption, processor):
+    result_md5_list = []
+    max_rss_mem_consumption = ''
+    max_uss_mem_consumption = ''
+    max_shared_mem_consumption = ''
+    if (args['mem_consumption'] == 1 and num == 0) or args['mem_consumption'] == 2:
+        mem_consumption.start_collect_memory_consumption()
+    start = time.perf_counter()
+    result_text = pipe.generate(
+        raw_speech,
+        max_new_tokens=1000,
+        # 'task' and 'language' parameters are supported for multilingual models only
+        language="<|en|>",
+        task="transcribe",
+        return_timestamps=True
+    )
+    end = time.perf_counter()
+    tm_list = np.array(result_text.perf_metrics.raw_metrics.m_durations) / 1000 / 1000
+    result_text = result_text.texts[0]
+
+    generation_time = end - start
+    out_data = processor.tokenizer(result_text, return_tensors='pt')
+    out_data.pop('token_type_ids', None)
+    # Remove `token_type_ids` from inputs
+    out_tokens = out_data['input_ids'] if 'input_ids' in out_data else out_data
+    out_token_size = out_tokens[0].numel()
```

Comment:
probably better to use openvino tokenizer from whisper pipeline for that instead of hf (some differences in tokenizer behaviour maybe found), we did not use it early due to issues with windows package, but now it should be resolved and you can use genai tokenizer directly

Reply:
@eaidova Do you know how to use genai tokenizer? I found whisper pipeline has the function get_tokenizer(), but it seems does not support obtaining the token size of the result text.  Such as implementing this function:  out_data = processor.tokenizer(result_text, return_tensors='pt').  We need to output the token size of the result text to CSV and JSON file.

Reply:
ok, I'll provide code snippet soon

Reply:
this part requires refactoring in text generation too, so I'll update both pipelines when this one will be merged

Reply:
@eaidova  Just to make sure I understand correctly, will the refactoring will fix following issue?
Output token size is much more than inference number of decoding model. 
Output size is the size of processor.tokenizer(result_text.texts[0]) and len of tokens equals to inference number
```
[ INFO ] [warm-up][P0] Output size: 59, Generation Time: 0.40s,
[ INFO ] [warm-up][P0] First token latency: 99.98 ms/token, other tokens latency: 3.61 ms/token, len of tokens: 78 * 1
```

Reply:
not fully understand what you mean, can you provide more details what you mean.

Output token size is much more than inference number of decoding model. - does it come from (which part of code calculate that)?

Reply:
@eaidova 
Output token size is from
https://github.com/openvinotoolkit/openvino.genai/blob/1fb7cafa0120eab72c5cd4bebd88f345f9e7901d/tools/llm_bench/task/speech_to_text_generation.py#L73
 
len of tokens is from length of
https://github.com/openvinotoolkit/openvino.genai/blob/1fb7cafa0120eab72c5cd4bebd88f345f9e7901d/tools/llm_bench/task/speech_to_text_generation.py#L53
or
https://github.com/openvinotoolkit/openvino.genai/blob/1fb7cafa0120eab72c5cd4bebd88f345f9e7901d/tools/llm_bench/task/speech_to_text_generation.py#L64

from the log:
[ INFO ] [warm-up][P0] **Output size: 59**, Generation Time: 0.40s,
[ INFO ] [warm-up][P0] First token latency: 99.98 ms/token, other tokens latency: 3.61 ms/token, **len of tokens: 78** * 1

Output token size is less than len of tokens.